usecases/schema/tenant.go

```diff
@@ -170,6 +206,139 @@ func (m *Manager) onAddTenants(ctx context.Context, class *models.Class, request
 	return nil
 }
 
+// UpdateTenants is used to set activity status of tenants of a class.
+//
+// Class must exist and has partitioning enabled
+func (m *Manager) UpdateTenants(ctx context.Context, principal *models.Principal,
+	class string, tenants []*models.Tenant,
+) error {
+	if err := m.Authorizer.Authorize(principal, "update", tenantsPath); err != nil {
+		return err
+	}
+
+	names := make([]string, len(tenants))
+	statuses := make([]string, len(tenants))
+	for i, tenant := range tenants {
+		names[i] = tenant.Name
+		statuses[i] = tenant.ActivityStatus
+	}
+
+	// validation
+	if err := validateTenants(names); err != nil {
```

Comment:
`validateTenants` could be refactored to accept `[]*models.Tenant` and validate the `tenant.Name`, rather than making extra allocations for `names`

Reply:
`validateTenants` is used also in `DeleteTenants` where only strings are given as input.
I can create 2nd method to validate both tenant names and activity statuses at once to be used in create and update methods.

Reply:
`AddTenants` needs names as strings to get partitions, so they have to be extracted from tenant struct anyway. Therefore new validation method would be used only for update case. I will stick with current impl for now.